students-router.js

```diff
@@ -0,0 +1,88 @@
+const knex = require('knex');
+const knexConfig = require('./knexfile.js');
+const db = knex(knexConfig.development);
+
+const router = require('express').Router();
+
+router.post('/', async (req,res) => {
+    if (!req.body.name || !req.body.cohort_id) {
+        res.status(400).json({error:'missing name / cohort id'});
+    } else {
+        try {
+            const student = await db('students').insert(req.body);
+            res.status(200).json(student);
+        }
+        catch(err) {
+            res.status(500).json({error:err});
+        }
+    }
+
+});
+
+router.get('/', async (req,res) => {
+    try {
+        const students = await db('students');
+        res.status(200).json(students);
+    }
+    catch(err) {
+        res.status(500).json({error:err});
+    }
+});
+
+router.get('/:id', async (req,res) => {
+    const {id} = req.params;
+    try {
+        const student = await
+        db.select('students.id','students.name','cohorts.name as cohort').from('students')
```

Comment:
good job on this stretch query, it was deceptive because without assigning a new reference to the cohort.name it would be adding two `name` properties to the same object, and the most recent one would override the first since an object can't have two properties with the same name :+1: